lighthouse-cli/bin.ts

```diff
@@ -35,10 +35,12 @@ import * as Printer from './printer';
 import * as randomPort from './random-port';
 import {Results} from './types/types';
 const yargs = require('yargs');
-const opn = require('opn');
-const updateNotifier = require('update-notifier');
 const pkg = require('../package.json');
 
+// accept noop modules for these, so the real dependency is optional.
+const opn = require('./shim-modules').opn();
```

Comment:
oooooh not literal `optionalDependencies` just handle them optionally :) I gotcha now

is there any benefit to having these be individual functions vs just inlining the exports? feels a little weird as functions